eng/Versions.props

```diff
@@ -137,11 +137,12 @@
     <Tooling_MicrosoftCodeAnalysisAnalyzersPackageVersion>3.3.4</Tooling_MicrosoftCodeAnalysisAnalyzersPackageVersion>
     <MicrosoftCodeAnalysisExternalAccessRazorPackageVersion>$(RoslynPackageVersion)</MicrosoftCodeAnalysisExternalAccessRazorPackageVersion>
     <MicrosoftCodeAnalysisExternalAccessOmniSharpCSharpPackageVersion>$(RoslynPackageVersion)</MicrosoftCodeAnalysisExternalAccessOmniSharpCSharpPackageVersion>
-    <MicrosoftCodeAnalysisCommonPackageVersion>$(RoslynPackageVersion)</MicrosoftCodeAnalysisCommonPackageVersion>
-    <MicrosoftCodeAnalysisCSharpPackageVersion>$(RoslynPackageVersion)</MicrosoftCodeAnalysisCSharpPackageVersion>
+    <!-- Microsoft.CodeAnalysis.Common and Microsoft.CodeAnalysis.CSharp packages are consumed in source-build.
+         Package version has to match roslyn repo reference as specified in MicrosoftNetCompilersToolsetVersion -->
+    <MicrosoftCodeAnalysisCommonPackageVersion>$(MicrosoftNetCompilersToolsetVersion)</MicrosoftCodeAnalysisCommonPackageVersion>
+    <MicrosoftCodeAnalysisCSharpPackageVersion>$(MicrosoftNetCompilersToolsetVersion)</MicrosoftCodeAnalysisCSharpPackageVersion>
     <MicrosoftCodeAnalysisCSharpEditorFeaturesPackageVersion>$(RoslynPackageVersion)</MicrosoftCodeAnalysisCSharpEditorFeaturesPackageVersion>
     <MicrosoftCodeAnalysisCSharpFeaturesPackageVersion>$(RoslynPackageVersion)</MicrosoftCodeAnalysisCSharpFeaturesPackageVersion>
-    <MicrosoftCodeAnalysisCSharpPackageVersion>$(RoslynPackageVersion)</MicrosoftCodeAnalysisCSharpPackageVersion>
     <MicrosoftCodeAnalysisCSharpWorkspacesPackageVersion>$(RoslynPackageVersion)</MicrosoftCodeAnalysisCSharpWorkspacesPackageVersion>
     <MicrosoftCodeAnalysisEditorFeaturesCommonPackageVersion>$(RoslynPackageVersion)</MicrosoftCodeAnalysisEditorFeaturesCommonPackageVersion>
     <MicrosoftCodeAnalysisEditorFeaturesTextPackageVersion>$(RoslynPackageVersion)</MicrosoftCodeAnalysisEditorFeaturesTextPackageVersion>
```

Comment:
These all need to be the same version. The likelihood is really that we can entirely remove `MicrosoftNetCompilersToolsetVersion` and just make everything `RoslynPackageVersion`, and then update from source-build as needed. I'm still not super comfortable with that, but I think that if we move this version to being auto-updated by darc on a regular cadence that would ameliorate much of my concern about shipping a different product in source-build and everywhere else.

Reply:
> These all need to be the same version. The likelihood is really that we can entirely remove `MicrosoftNetCompilersToolsetVersion` and just make everything `RoslynPackageVersion`, and then update from source-build as needed. I'm still not super comfortable with that, but I think that if we move this version to being auto-updated by darc on a regular cadence that would ameliorate much of my concern about shipping a different product in source-build and everywhere else.

Source-build does not know about repo-specific meta properties like `RoslynPackageVersion`. Source-build needs to update versions of all packages consumed during-source build, to ensure that versions are correct and all dependent packages are available in offline package cache.

@mmitche is there anything we haven't tried to reconcile the differing requirements by razor's regular- and source- builds? We could set some of these properties differently, only in source-build, i.e. `MicrosoftCodeAnalysisCommonPackageVersion` and `MicrosoftCodeAnalysisCSharpPackageVersion`.

Reply:
Right, agreed. So I think you basically want to:
- Remove RoslynPackageVersion, since it can't be updated by Maestro. Today this is updated manually.
- Instead, use the properties that correspond to all the individual packages, and simply set their versions to be all identical. All of these dependencies should go into Version.Details.xml, or else source build will not properly override them with PVP flow.
- Then, @333fred and his team decide on the build of roslyn that gets used in this repo. An update just sets all the package versions. That could either be done via darc update-dependencies manually, or via regular updates from roslyn through Maestro.

So, you get the same behavior today you have with `RoslynPackageVersion`, a predictable version. It just also gets used to compile the repo via MicrosoftNETCompilersToolsetPackageVersion.

Thoughts?

Reply:
That sounds good. We've been talking about having things flow through Maestro (to eliminate the problem of source-build building a product we never test), but that setup will probably wait until after I'm back from vacation. For right now, the version being used in `RoslynPackageVersion` should work.

Reply:
Pushed the changes to address this: https://github.com/dotnet/razor/pull/8492/commits/647f22f823360dee2d3cb0f7bd1d10d0084d1a6e. I think it should be ok now.

Reply:
Code-check is clean now: https://github.com/dotnet/razor/pull/8492/commits/1f8ba5ecd72b5da928f7316fd88de7adb10a472e

Reply:
@333fred @mmitche I implemented the changes as discussed above. Can you take another look and approve?